src/content/popup/KanjiReferencesTable.tsx

```diff
@@ -0,0 +1,115 @@
+import { KanjiResult } from '@birchill/jpdict-idb';
+
+import { ReferenceAbbreviation, getSelectedReferenceLabels } from '../../common/refs';
+import { getReferenceValue } from '../reference-value';
+import { I18nProvider, useLocale } from '../../common/i18n';
+import { useMemo } from 'preact/hooks';
+
+type Props = {
+  entry: KanjiResult,
+  kanjiReferences: Array<ReferenceAbbreviation>,
+}
+
+// temporary hack
+export function WrappedKanjiReferencesTable(props: Props) {
+  return (
+    <I18nProvider>
+      <KanjiReferencesTable {...props} />
+    </I18nProvider>
+  )
+}
+
+export function KanjiReferencesTable({ entry, kanjiReferences }: Props) {
+  const { t, langTag } = useLocale();
+
+  const {
+    gridAutoFlow,
+    gridTemplateRows,
+    referenceTableInfo
+  } = useMemo(() => {
```

Comment:
This might be a little neater if we continue calculating `gridAutoFlow` and `gridTemplateRows` outside the `useMemo` hook so that the `useMemo` hook is simply responsible to producing the `referenceTableInfo` value.